### ISSUE TITLE
🛠 Repo: Set up Renovate

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [mochajs/mocha#5055]
### Tooling Suggestion Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `master` branch of the repository.
- [X] I have read and agree to Mocha's [Code of Conduct](https://github.com/mochajs/mocha/blob/master/.github/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/mochajs/mocha/blob/master/.github/CONTRIBUTING.md)
- [X] I have searched for [related issues](https://github.com/mochajs/mocha/issues?q=is%3Aissue) and [issues with the `faq` label](https://github.com/mochajs/mocha/issues?utf8=%E2%9C%93&q=is%3Aissue%20label%3Afaq%20), but none matched my issue.
- [X] I want to provide a PR to resolve this

### Overview

There are quite a few issues on this repo right now tracking out-of-date packages. We're going to enable [Renovate](https://github.com/renovatebot/renovate) to automatically create PRs to update packages.

Notes from @boneskull:

> Looks reasonable. The only thing I’d suggest with the automated tooling (something like Renovate, yeah?) is to have a human in the loop for production dependency upgrades—check them closely. Mocha has been bitten several times due to semver violations in its dependency tree. I would also recommend using https://socket.dev/ as well (should be free for OSS). While Mocha hasn’t been hit by a malicious dep (to my knowledge), historical performance is not a guarantee of future results. 😄

**Comment by JoshuaKGoldberg** [comment]
We'd looked at this a few times last year. Unfortunately, the old docdash-based website has a ton of outdated dependencies that make this rather difficult. If we were to set up Dependabot/Renovate today then we'd have to exclude a heck of a lot of dependencies from it.

#5207 -> #5246 tracks making a new website on a more modern stack (Astro). Once the new website fully replaces the old one then we'll be free of the ancient docdash dependencies. In the meantime, marking this as blocked. 🛑

**Comment by mark-wiemer** [comment]
Assigning myself to track this one, I'm OK excluding dependencies in the short-term if it means at least some automated upgrade PRs are made available :) @JoshuaKGoldberg OK marking as unblocked? We can update the "archive old site" issue to say "and also cleanup the Renovate ignore list" or similar

**Comment by JoshuaKGoldberg** [comment]
That sounds reasonable to me.

...or even better, how about we just go ahead and move the legacy docs site over now? Are there any blockers?

**Comment by mark-wiemer** [comment]
@JoshuaKGoldberg I don't see any blockers in moving the code contents to a new repo aside from making sure the GitHub Actions / Netlify hooks work with the new site. Here's how I would do it:

1. Copy old site code into new repo (already done, I believe)
1. Deploy old site from new repo to make sure it works (already done?)
1. Remove old site code from current repo

Tada :)

**Comment by JoshuaKGoldberg** [comment]
Swell. #5583 was already in draft; I cleaned it up, finished adding redirects, and un-drafted it just now.

**Comment by mark-wiemer** [comment]
merged!

**Comment by JoshuaKGoldberg** [comment]
I went ahead and enabled Renovate on this repo. https://developer.mend.io/github/mochajs/mocha

**Comment by mark-wiemer** [comment]
Cool, linking #5639

**Comment by mark-wiemer** [comment]
@JoshuaKGoldberg can we close this one?

**Comment by mark-wiemer** [comment]
I'm going rogue and closing this one :D

**Comment by mark-wiemer** [comment]
Ref #5704
